### PR TITLE
Feat js t2001

### PR DESCRIPTION
**Pull request by sganeshkumaar**

### CODE REVIEW

**Comment by lakshmipooja1510** [comment]
image
On Scrolling the dropdown is also getting scrolled. Can u work on this bug

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
+/**
+ * The json data fetched from data.json
+ */
+let coreData
+
+/* jutky */
```
Comment (lakshmipooja1510) [inline src/index.js:6]: What does this comment mean?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+/**
+ *
+ * @param {string} city city-name
+ * fills time and date
+ */
+function fillTopTimeStamp (city) {
+  let [date, time, state] = coreData[city].dateAndTime.split(' ')
+  date = date.slice(0, -1)
```
Comment (lakshmipooja1510) [inline src/index.js:73]: Can u rename state as something like timePeriod etc. As it is misleading

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  const [month, day, year] = date.split('/')
+  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+  const dateString = day + '-' + months[month - 1] + '-' + year
+  document.getElementsByClassName('forcast-date')[0].innerHTML = dateString
+}
+
+/**
+ *
+ * @param {string} time time in a : seperated format hr:min:sec
+ * @param {string}state am or pm
```
Comment (lakshmipooja1510) [inline src/index.js:93]: image
The spelling for separated is wrong I guess. So it is showing a blue underline. Please ensure it is cleared

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+ * fills hours , minutes ,seconds and state forcast-time class
+ */
```
Comment (lakshmipooja1510) [inline src/index.js:95]: Check spelling for forecast

File: src/index.js (hunk 1 above)
Comment (sganeshkumaar) [reply src/index.js:6]: i think i must have added it by mistake.

File: src/index.js (hunk 3 above)
Comment (sganeshkumaar) [reply src/index.js:93]: sure , ill make sure there isnt any spelling mistakes in the code

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  }
+}
+
+/**
+ *
+ * @param {string} city city-name
+ * fills temperature in C, F, humidity and precipitation %
+ */
+function fillTopFactors (city) {
+  const tempInCelcius = parseInt(coreData[city].temperature.slice(0, -2))
```
Comment (lakshmipooja1510) [inline src/index.js:120]: Check spelling for celsius

File: src/index.js (hunk 4 above)
Comment (sganeshkumaar) [reply src/index.js:95]: ill change this name in each html,css and js file

**Comment by sganeshkumaar** [comment]
> image On Scrolling the dropdown is also getting scrolled. Can you work on this bug


its not happening for me .  it happens only if i  inspect the site. also it happens to the site you gave for reference
image

File: src/index.js (hunk 2 above)
Comment (sganeshkumaar) [reply src/index.js:73]: ok ill rename it to timePeriod

**Review by lakshmipooja1510 — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+  for (let i = 0; i < 6; i++) {
+    timelineBox[i].getElementsByClassName('timeline-time')[0].innerHTML = 'NIL'
+    timelineBox[i].getElementsByClassName('timeline-icon')[0].firstElementChild.src = '../Assets/HTM&CSS/General_Images_&_Icons/nil.svg'
+    timelineBox[i].getElementsByClassName('timeline-icon')[0].firstElementChild.style = 'height:21px; filter: grayscale(100%);'
+    timelineBox[i].getElementsByClassName('timeline-temp')[0].innerHTML = 'NIL'
+  }
+}
+/* adding load eventlistener on load */
+window.addEventListener('load', loadFuncion)
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:233]: that spelling mistake has been resolved

File: src/index.js (hunk 5 above)
Comment (sganeshkumaar) [reply src/index.js:120]: changed to the right spelling

**Pull request merged**